data_transfer/utils/__init__.py

```diff
@@ -47,3 +49,40 @@ def read_json(filepath: Path) -> Any:
 def write_json(filepath: Path, data: dict) -> None:
     with open(filepath, "w") as f:
         json.dump(data, f, indent=4)
+
+
+def validate_and_format_patient_id(ideafast_id: str) -> Union[bool, str]:
+    """
+    Validate a (messy) IDEAFAST id, based on ideafast/ideafast-idgen
+    Returns boolean if validate with corrected formatting
+    """
+
+    if type(ideafast_id) is str:
+        id_without_punc = re.sub(r"[^\w]", "", ideafast_id)
+        #  TODO: remove spaces if present
```

Comment:
Potentially also remove tabs ("\t") that may be added historically to the inventory?

```python
id_without_punc = id_without_punc.replace(" ", "").replace("\t", "")
```

It would be worth adding another unit test to validate this too

Reply:
The `^\w` regex already identifies (and remove) spaces and tabs, but adding tests to prove that is a very good point.

Reply:
Actually, it does include an underscore - I suppose we do want to remove those (however unlikely they are in there).  Then it should be:
```python
 id_without_punc = re.sub(r"[^\w]|_", "", ideafast_id)
```

Reply:
Whoops -- My mistake! The `test_punctuation` test verifies no spaces, although potentially having an explicit test for tabs/spaces might be good anyway?